src/main/scala/amailp/intellij/robot/psi/LibraryValue.scala

```diff
@@ -5,12 +5,17 @@ import com.intellij.extapi.psi.ASTWrapperPsiElement
 import com.intellij.lang.ASTNode
 import com.intellij.psi.PsiReference
 import com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry
+import com.intellij.psi.util.QualifiedName
 
 class LibraryValue(node: ASTNode) extends ASTWrapperPsiElement(node) with RobotPsiUtils with Library {
-
+  private val LIBRARY_SYNONYM_SEPARATOR: String = "WITH NAME"
   def getType: String = "Library"
 
-  override def getText: String = getNode.getText
+  def getRobotName: String = getNode.getText.split(LIBRARY_SYNONYM_SEPARATOR).last.trim
+
+  def getQualifiedName: QualifiedName = QualifiedName.fromDottedString(getText)
+
+  override def getText: String = getNode.getText.split(LIBRARY_SYNONYM_SEPARATOR).head.split(' ').head.trim
```

Comment:
second split to remove class arguments. don't like how it's done

Reply:
Keepinng the previous implementation in `SettingsParser` here we could probably avoid all the string splitting that would be done by the parser and instead do the analysis of the `PsiElement` tree.